src/core/map.cpp

```diff
@@ -2061,8 +2061,8 @@ void Map::removePart(std::size_t index)
 	Q_ASSERT(parts.size() > 1);
 	
 	if (current_part_index == index)
-		// First switch to another part when removing the current part
-		setCurrentPartIndex((index == parts.size() - 1) ? (parts.size() - 2) : (index + 1));
+		// First switch to another part when removing the current part, this will also clear any object selection
+		setCurrentPartIndex((index == parts.size() - 1) ? (index - 1) : (index + 1));
```

Comment:
Although `index - 1` is equivalent to `parts.size() - 2` I prefer it due to readability.